### PR TITLE
Added a logger rewriter for correctly serialize metadata objects.

### PR DESCRIPTION
**Pull request by Seldszar**
The following change allows objects without a `toJSON` method being correctly serialized. For example, an `Error` object will return `{}` instead of showing the various properties (name, message, stack, etc), making the logging entry unhelpful.

The algorithm will walk through prototypes to collect all the inherited properties and build a plain object with them.

### CODE REVIEW

**Comment by Lange** [comment]
Neat! Can you add a test for this new functionality?

**Comment by Seldszar** [comment]
I can, sure!

File: test/logger/server.js [hunk 1]
```diff
@@ -140,6 +141,33 @@ test('logging methods should generate any output when of an adequate level', t =
 	process.stdout.write.restore();
 });
 
+test('should not output an empty object', t => {
```
Comment (Lange) [inline test/logger/server.js]: There's times where it can be useful to know that something is an empty object. This would hide that from the logs.

Comment (Seldszar) [reply test/logger/server.js:144]: I see, I should check if metadata is `undefined` then.

Comment (Seldszar) [reply test/logger/server.js:144]: Unfortunately, that's impossible to do something like that. The `meta` will always be an object whatever the case.

So I have two options:
- Write the meta in both console and file
- Only write the `meta` object in the log file, and hide it on console

Which one would be better for you?

Comment (Seldszar) [reply test/logger/server.js:144]: Or maybe it's possible by moving the serialization to the `Logger` class methods. I'll try this way.

So, the challenge with this solution will be to recreate the whole interpolation algorithm but change the `meta` object. In this case, we can determine if an empty object was set or not.

**Comment by Lange** [comment]
It looks like `rewriters` have been removed from winston 3, so this PR likely needs to be refactored: https://github.com/winstonjs/winston/issues/1140

**Comment by Seldszar** [comment]
Agreed, I was recently refactoring it on my side, which basically includes the PR features with additional ones. Maybe this one can be used instead?

https://github.com/Seldszar/nodecg/tree/feature/logger

**Comment by Lange** [comment]
Sure, I generally agree that NodeCG's logger could use some work, and it
looks like you've put a lot of thought into it. I'd be okay with closing
this PR and opening another one with your new changes, if that sounds good
to you.

**Comment by Seldszar** [comment]
Sure thing!

**Pull request closed without merging**